ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -640,4 +574,136 @@ private void onAdded(final Transaction transaction) {
       onAddedListeners.forEach(listener -> listener.onTransactionAdded(transaction));
     }
   }
+
+  class SaveRestoreManager {
+    private final Lock diskAccessLock = new ReentrantLock();
+    private final AtomicReference<CompletableFuture<Void>> writeInProgress =
+        new AtomicReference<>(CompletableFuture.completedFuture(null));
+    private final AtomicReference<CompletableFuture<Void>> readInProgress =
+        new AtomicReference<>(CompletableFuture.completedFuture(null));
+    private final AtomicBoolean isCancelled = new AtomicBoolean(false);
+
+    void saveToDisk(final PendingTransactions pendingTransactionsToSave) {
+      serializeAndDedupOperation(
+          () -> executeSaveToDisk(pendingTransactionsToSave), writeInProgress);
+    }
+
+    void loadFromDisk() {
+      serializeAndDedupOperation(this::executeLoadFromDisk, readInProgress);
+    }
+
+    private void serializeAndDedupOperation(
+        final Runnable operation,
+        final AtomicReference<CompletableFuture<Void>> operationInProgress) {
+      if (configuration.getEnableSaveRestore()) {
+        diskAccessLock.lock();
```

Comment:
This call to lock should use `tryLock(long time, TimeUnit unit)` to ensure it doesn't wait forever getting a lock.

Reply:
done